### PR TITLE
feat: add `includeSources` to feed settings

### PR DESCRIPTION
**Pull request by ilasw**
## Changes
- Added `includeSources ` field in feed settings;
- Added mutation logic;
- Updated feed tests for new feature;

### CODE REVIEW

File: src/entity/FeedSource.ts
Comment (capJavert) [inline src/entity/FeedSource.ts:16]: maybe better to make it enum with `included` and  `excluded` or whatever naming. you want, so we can expand more easily?

Comment (rebelchris) [reply src/entity/FeedSource.ts:16]: This follows the tag approach and we thought about it, but you can actually only "block" or "unblock" (neutral meaning removal) so would keep this for now atleast. (as it's only a temporary solution till we get experiment results and we know wheter to go full blown with this)

Comment (capJavert) [reply src/entity/FeedSource.ts:16]: Ok, not really in the loop what it should done.

**Review by rebelchris — APPROVED** [review]
Looks good to me, but wait for Ante's approval as well please 🙏

File: src/entity/FeedSource.ts
Comment (capJavert) [reply src/entity/FeedSource.ts:16]: @ilasw do you have frontend PR which uses this, just would like to see how it works as well before merging?

Comment (ilasw) [reply src/entity/FeedSource.ts:16]: Hi @capJavert , yes! You can use is pulling from this branch: https://github.com/dailydotdev/apps/pull/3360 both in source page and in post article.

Comment (capJavert) [reply src/entity/FeedSource.ts:16]: tnx, gonna check it out!

**Review by sshanzel — APPROVED** [review]
Looks amazing! One question, would we get another PR for updating user preference on whether to receive new Squad post notifications if they were previously subscribed?

File: src/schema/feeds.ts
Comment (ilasw) [reply src/schema/feeds.ts]: @capJavert tried but cannot revert, IDE auto format this way even if copy-paste from old source or rebase from main. There is some config that I have to enable/change?

**Comment by ilasw** [comment]
> would we get another PR for updating user preference on whether to receive new Squad post notifications if they were previously subscribed?
Currently there is no reference for evolutions on the API side, not sure to understand this behavior!

**Comment by sshanzel** [comment]
> > would we get another PR for updating user preference on whether to receive new Squad post notifications if they were previously subscribed?
> 
> Currently there is no reference for evolutions on the API side, not sure to understand this behavior!

When users follow and subscribe to a source, we store some information in the database. As a user, if I end up blocking a source that I subscribed to before, I expect to not receive notifications from them, and that information was the one stored in the database.

File: src/schema/feeds.ts
Comment (capJavert) [reply src/schema/feeds.ts]: do you have some other formatter enabled in your IDE, because eslint/prettied that is setup in project does not format that part.

**Comment by ilasw** [comment]
> When users follow and subscribe to a source, we store some information in the database. As a user, if I end up blocking a source that I subscribed to before, I expect to not receive notifications from them, and that information was the one stored in the database.

@sshanzel  I've tested this behavior, and when you block a source, you will no longer receive notifications. It seems to have been like this even before this PR!

**Comment by sshanzel** [comment]
> > When users follow and subscribe to a source, we store some information in the database. As a user, if I end up blocking a source that I subscribed to before, I expect to not receive notifications from them, and that information was the one stored in the database.
> 
> @sshanzel I've tested this behavior, and when you block a source, you will no longer receive notifications. It seems to have been like this even before this PR!

Nice. Thank you for confirming 🚀

**Pull request merged**
